raiden/raiden_service.py

```diff
@@ -95,10 +95,17 @@
 # register filelock logger
 filelock.logger = slogging.get_logger('filelock')
 
+TransferResult = namedtuple('TransferResult', (
+    'async_result',
+    'amount',
+    'token',
+    'target',
+))
+
 
 def create_default_identifier():
     """ Generates a random identifier. """
-    return random.randint(0, UINT64_MAX)
+    return int(time.time()) * 1000 + random.randint(0, 999)
```

Comment:
what is the point of this change? a random identifier is a random identifier, I prefer the simpler version.

Reply:
The point here is that this PR isn't going to go allow `identifier` collision, not only in each channel, but across every transfer. This change makes the identifier go monotonically growing (on time), besides giving some information on when the call was made (timestamp based). It's a common pattern on UI development which depends on unique identifier generation, and as I plan to implement the same behavior on the webUI, I thought it could be useful to stay consistent and use it here too.

Reply:
I don't agree with the reasoning behind this change:

> The point here is that this PR isn't going to go allow identifier collision, not only in each channel, but across every transfer.

The old approach works as fine.

> This change makes the identifier go monotonically growing (on time), 

Not true, since the value has a piece of randomness in it, also I don't see what is the value of a monotonic identifier, specially since we have the nonce.

> besides giving some information on when the call was made (timestamp based)

What is the benefit?

Reply:
> Not true, since the value has a piece of randomness in it, also I don't see what is the value of a monotonic identifier, specially since we have the nonce.

It would only not me monothonic if it was ran more than once a second, which isn't expected (see that the random part is the last 3 digits, _random_ milliseconds, real seconds)). Nonce doesn't give time info, only order.

> What is the benefit?

The benefit is adding an information to the identifier (**when** the call started), without compromising the current behavior (random/non-collidable identifier).

If the old approach works as fine as this one, we can make the change for the benefits above, or if not, at least for consistency with webUI behavior, which will have time-based identifier generation.

In any case, this method is called only on missing identifier, and, in the practice, only webUI-generated (time-based) identifier will be used. In the future, with this idempotency approach to the other state-changing methods, we may even require client-provided `identifier`, which will get this auxiliary function removed.

Reply:
> It would only not be monothonic if it was ran more than once a second, which isn't expected (see that the random part is the last 3 digits, random milliseconds, real seconds)).

So ... not monotonic.

> Nonce doesn't give time info, only order.

Correct.

I'm asking **why do you want to know the time a transfer started through it's identifier**? If there is no reason, please revert it to a random identifier.

Reply:
As you wish.